addons/product/models/product_product.py

```diff
@@ -308,34 +311,15 @@ def _compute_combination_indices(self):
     def _compute_is_product_variant(self):
         self.is_product_variant = True
 
-    @api.onchange('lst_price')
-    def _set_product_lst_price(self):
-        for product in self:
-            if self.env.context.get('uom'):
-                value = self.env['uom.uom'].browse(self.env.context['uom'])._compute_price(product.lst_price, product.uom_id)
-            else:
-                value = product.lst_price
-            value -= product.price_extra
-            product.write({'list_price': value})
-
     @api.depends("product_template_attribute_value_ids.price_extra")
     def _compute_product_price_extra(self):
         for product in self:
             product.price_extra = sum(product.product_template_attribute_value_ids.mapped('price_extra'))
 
     @api.depends('list_price', 'price_extra')
```

Comment:
(Another side-comment, can be handled after freeze, sorry for not noticing earlier)
Ideally, we should change the dependency here to `product_template_attribute_value_ids.price_extra`, as it's a bad practice to make stored computed field depend on non-stored computed fields (using the non stored one in the computation is okayish, but having it as dependency might trigger unwanted/useless recomputations)